feat: add HeaderConfig to SecurityConfig (#11485)
tauri-apps/tauri: .changes/add-headers-config-option.md

---
"tauri-utils": 'minor:feat'
"tauri": 'minor:feat'
---

> Adds a new configuration option `headers` under >app>security for the tauri configuration file. Headers defined there are added to every http response from tauri to the web view. This doesn't include IPC messages and error responses. The `Content-Security-Policy`(CSP) remains untouched and still is defined separately.

Reviewer: ```suggestion
Adds a new configuration option `app > security > headers` to define headers that will be added to every http response from tauri to the web view. This doesn't include IPC messages and error responses.
```

---

> "@tauri-apps/api": 'minor:feat'
> "tauri-utils": 'minor:feat'
> "tauri-macos-sign": 'minor:feat'
> "tauri-bundler": 'minor:feat'
> "tauri-runtime": 'minor:feat'
> "tauri-runtime-wry": 'minor:feat'
> "tauri-codegen": 'minor:feat'
> "tauri-macros": 'minor:feat'
> "tauri-plugin": 'minor:feat'
> "tauri-build": 'minor:feat'
> "tauri": 'minor:feat'
> "@tauri-apps/cli": 'minor:feat'
> "tauri-cli": 'minor:feat'
> "tauri-driver": 'minor:feat'

Reviewer: This change only concerns `tauri-utils` and `tauri`
```suggestion
"tauri-utils": 'minor:feat'
"tauri": 'minor:feat'
```

```http
access-control-allow-origin:  http://tauri.localhost
access-control-expose-headers: Tauri-Custom-Header
content-security-policy: default-src 'self'; connect-src ipc: http://ipc.localhost; script-src 'self' 'sha256-Wjjrs6qinmnr+tOry8x8PPwI77eGpUFR3EEGZktjJNs='
content-type: text/html
cross-origin-embedder-policy: require-corp
cross-origin-opener-policy: same-origin
tauri-custom-header: key1 'value1' 'value2'; key2 'value3'
timing-allow-origin: https://developer.mozilla.org, https://example.com
```
Since the resulting header values are always 'string-like'. So depending on the what data type the HeaderSource is, they need to be converted.
 - `String`(JS/Rust): stay the same for the resulting header value
 - `Array`(JS)/`Vec\<String\>`(Rust): Item are joined by ", " for the resulting header value

>  - `Object`(JS)/ `Hashmap\<String,String\>`(Rust): Items are composed from: key + space + value. Item are then joined by "; " for the resulting header value

Reviewer: This is a bit verbose for a change log entry, let's reword into 2~5 lines at max, examples and extensive documentation should be in the inline documentation
